### PR TITLE
Image size access from ImageAnnotation

### PR DESCRIPTION
**Pull request by hepengfe**
This PR fixes https://github.com/asyml/forte/issues/880.

### Description of changes
* implement a function in image annotation to get the image size by `ImageAnnotation.pack.get_payload().cache_shape`

### Possible influences of this PR.
* `ImageAnnotation` can access image size that it operates on.
* all payloads can be initialized with data shape
    For text data, it is the length of the text.[length, text_embedding_dim]
    For audio data, it is the length of the audio. [length, audio_embedding_dim]
    For image data, it is the shape of the image. [height, width, channel]

### Test Conducted
Describe what test cases are included for the PR.

### CODE REVIEW

File: forte/data/ontology/top.py [hunk 1]
```diff
@@ -883,26 +883,41 @@ def image(self):
...
-            height
-        )
+        if self.pack is None:
+            raise ValueError(
+                "Cannot get image because image annotation is not "
+                "attached to any data pack."
+            )
+        image_shape = self.pack.get_payload_at(
+            Modality.Image, self._image_payload_idx
+        )._cache_shape
```
Comment (mylibrar) [inline forte/data/ontology/top.py:913]: Maybe it's better to use `image_shape` here.

File: forte/data/ontology/top.py [hunk 2]
```diff
@@ -1219,15 +1235,36 @@ def payload_index(self) -> int:
...
+                on the modality and its length.
+                For text data, it is the length of the text.[length, text_embedding_dim]
+                For audio data, it is the length of the audio. [length, audio_embedding_dim]
+                For image data, it is the shape of the image. [height, width,
+                channel]
+                For example, for image data, if the cache_shape length is 2, it
+                means the image data is a 2D image [height, width].
         """
         self._cache = data
+        if isinstance(data, np.ndarray):
```
Comment (mylibrar) [inline forte/data/ontology/top.py:1263]: Looks like we can also set `cache_shape` when `data` is a string.

File: forte/data/ontology/top.py (hunk 1 above)
Comment (hepengfe) [reply forte/data/ontology/top.py:913]: do you mean change the variable name? `cache_shape` -> `image_shape`. `cache_shape` is more generic variable for payloads of all modalities.

Comment (mylibrar) [reply forte/data/ontology/top.py:913]: No you don't need to change the variable name. I think `Payload` already exposed a property function called `image_shape` right?

File: forte/data/ontology/top.py (hunk 2 above)
Comment (hepengfe) [reply forte/data/ontology/top.py:1112]: fixed it.

File: docs/notebook_tutorial/ocr.ipynb [hunk 3]
```diff
@@ -80,7 +89,8 @@
     "        # Read in Image data and store in DataPack\n",
     "        # add Image payload into DataPack.payloads\n",
     "        image_data = Image.open(requests.get(data_source, stream=True).raw)\n",
-    "        pack.set_image(image_data, payload_idx)\n",
+    "        width, height = image_data.size\n",
+    "        ip.set_cache(image_data, [height, width])\n",
```
Comment (mylibrar) [inline docs/notebook_tutorial/ocr.ipynb:93]: Do we need to explicitly set the shape here? `set_cache` can automatically infer the image shape when the input data is `ndarray`.

**Review by mylibrar — APPROVED** [review]

**Pull request merged**